### ISSUE TITLE
[Controller Request] : Y3 Joypad (Receiver YC5)

### ISSUE THREAD

**Issue by MJr0019** [o0Zz/sys-con#30]
Controller Name:

Y3 Joypad (Receiver YC5)

LINK: https://aliexpress.com/item/1005005458136372.html

Current Controller Behaviour:

Controller is not successful in pairing with the console

Context:

1- I connected the usb dongle to the USB to usb Type C Connector.

2- It was not connected.

**Comment by MJr0019** [comment]
I'm using this version of Sys-con:

https://github.com/o0Zz/sys-con/issues/19#issuecomment-2262066471

Here is the Log file:

[log.log](https://github.com/user-attachments/files/16464562/log.log)

Config file:

[config.ini.txt](https://github.com/user-attachments/files/16464561/config.ini.txt)

Wireshark Capture:

[Wireshark_Cap.zip](https://github.com/user-attachments/files/16464481/Wireshark_Cap.zip)

**Comment by o0Zz** [comment]
Ok this one is another story.

You got the same issue than this one: #21. Your controller is not detected at all by the nintendo switch. That's means sys-con has no chance to catch the controller. I'm very supprised this one is not reconized by the switch. According to the wireshark capture, everything is going well with this controller, there is not special things. 

When you say it's not connected, do you means the controller don't even light on ? (Not sure if these controller has a light?)

Could you confirm the vid/pid on windows is: [0810-0001] ? (To make sure I read the correct wireshark logs.)

So, let's try to play a bit with switch discovery and see if we can do somthing (I'm not really confident there ^^)

[sys-con-1.2.2+22-ATMOSPHERE-1.7.x.zip](https://github.com/user-attachments/files/16466996/sys-con-1.2.2%2B22-ATMOSPHERE-1.7.x.zip)

**Comment by MJr0019** [comment]
Hello friend.

_Q- When you say it's not connected, do you mean the controller doesn't even light on ? (Not sure if these controller has a light?)_ 

A- In this test I noticed that the control "connected" because the lights at the top stopped blinking, there was a fixed LED for each control.

In this test with the Y3 Joypad (Receiver YC5): 

Player 01 there will be a solid green LED.

Player 02 will have a solid red LED.

Q- _Could you confirm the vid/pid on windows is: [0810-0001] ? (To make sure I read the correct wireshark logs.)_

 A- I think so, because I removed all the USBs at the time of the test, leaving only the Y3 Joypad.

**BUT** how can I confirm this vid/pid?

Q- _So, let's try to play a bit with switch discovery and see if we can do something (I'm not really confident there ^^)_

A - should I change "discovery_mode=" and "log_level=" to different ?

Here is the log and config, all tests I did on this version:

[log.log](https://github.com/user-attachments/files/16470613/log.log)

[config.ini.txt](https://github.com/user-attachments/files/16470617/config.ini.txt)

**Comment by o0Zz** [comment]
Yes please change 
```
log_level=0
```
To confirm the PID -> https://github.com/o0Zz/sys-con/?tab=readme-ov-file#method-2-from-a-windows-pc

**Comment by MJr0019** [comment]
Here is the log and config:

[log.log](https://github.com/user-attachments/files/16474290/log.log)

[config.ini.txt](https://github.com/user-attachments/files/16474288/config.ini.txt)

Here is the VID/PID:

![image](https://github.com/user-attachments/assets/92e37eac-0d95-4be3-9f04-46bd15f6bb3f)

[0810-0001]

**Comment by o0Zz** [comment]
Test 1:
Try to plug your controller and to reboot the switch to see if during the boot the switch detect something.

Test 2:
Try to write this configuration in your [global]

```
log_level=0
discovery_mode=1
discovery_vidpid=0810-*
```

Let me know the result but I'm afraid to have reached a switch limitation :/

**Comment by MJr0019** [comment]
Hello friend !

Test 1:

[log_test01.log](https://github.com/user-attachments/files/16481082/log_test01.log)


[config_test01.ini.txt](https://github.com/user-attachments/files/16481081/config_test01.ini.txt)

Test 2:

[log_test02.log](https://github.com/user-attachments/files/16481083/log_test02.log)


[config_test02.ini.txt](https://github.com/user-attachments/files/16481084/config_test02.ini.txt)

In this two tests noticed that the control "connected" because the lights at the top stopped blinking, there was a fixed LED for each control.

In this  two tests with the Y3 Joypad (Receiver YC5):

Player 01 there will be a solid green LED.

Player 02 will have a solid red LED.

**Comment by o0Zz** [comment]
Ok thanks. So none of tests worked (from switch point of view) I assume at low level the switch skip this controller because the controller provide something the switch dont like. I will try to simulate your controller on my side with the wireshark capture you provided to check what parameter cause this issue and see if we can find a workaround. I will probavly need more time here. I keep you updated.

**Comment by MJr0019** [comment]
OK, thank you very much for your work!

**Comment by MJr0019** [comment]
Hello friend, any news?

Anything that might help?

**Comment by o0Zz** [comment]
Hi, I need to work deeper on this issue is the same than #21. I will try to simulate this controller in order to see if there is a workaround  but it will take some time. I will keep you updated.

**Comment by MJr0019** [comment]
Hello friend, any news?

Anything that might help?

**Comment by o0Zz** [comment]
Hi, 

Quick question, Do you use the Switch Dock ? I means, when you did these test, do you plug this controller to the dock or are you using a USB-C adapter ?
If you only tried Dock, could you try to use a USB-C adapter and see if this controller is working ?

Thanks

**Comment by MJr0019** [comment]
Hello Friend,

Q- A quick question, do you use the Switch Dock?

A- Yes

Q- I mean, when you did these tests, did you connect this controller to the dock or are you using a USB-C adapter?

A- I tested with both but I do more tests with the USB-C adapter because it is portable.

Q- If you only tried the Dock, could you try using a USB-C adapter and see if this controller is working?

A- In the last test I tried with the USB-C adapter, now I tested with the Dock here is your log:

[config.ini.txt](https://github.com/user-attachments/files/17215584/config.ini.txt)


[log.log.txt](https://github.com/user-attachments/files/17215587/log.log.txt)


Thanks for your support

**Comment by o0Zz** [comment]
Thank you for your reply. I did some investigation on my side in order to better understand the switch USB discovery process:
https://github.com/o0Zz/sys-con/blob/master/doc/ReverseEngineerUSB.md
For now I don't really understand what's wrong here.
Do you have any USB hub with an external power supply (Like this: https://www.amazon.co.uk/Powered-RSHTECH-Aluminum-4-port-Individual-Black/dp/B08ZKSK6MB)? Or do you have a way to have 1 in order to make a test ? I would like to check if the issue would be a "power consumption" issue.
In other words the idea would be to plug the powered USB hub to the switch and plug the dongle on it.

**Comment by MJr0019** [comment]
Q- Do you have any USB hub with an external power supply (Like this: https://www.amazon.co.uk/Powered-RSHTECH-Aluminum-4-port-Individual-Black/dp/B08ZKSK6MB)? Or do you have a way to have 1 in order to do a test? I would like to check if the issue would be a "power consumption" issue.

A- Yes, I have this one (https://down-br.img.susercontent.com/file/cn-11134207-7qukw-liihpnl2ewl6ac) 

I did 2 tests.

Test 1: 

[config.ini_TEST_01.txt](https://github.com/user-attachments/files/17371890/config.ini_TEST_01.txt)

[log.log_TEST_01.txt](https://github.com/user-attachments/files/17371892/log.log_TEST_01.txt)


The USB-C adapter + dongle of the Y3 Arcade Fighter Stick (DATA FROG) / Y3 Arcade Fighter Stick with GAME STICK 4K LITE SEGAM M8 V7.0 

foto

The USB-C adapter + dongle of the Y3 Joypad with Receiver no-name / Y3 Joypad with GAME STICK 4K LITE SEGAM M8 V4.0 

Foto

The USB-C adapter + Xbox One X/S Controller 

foto

The HUB USB 3.0 + USB-C adapter + Y3 Arcade Fighter Stick (DATA FROG) / Y3 Arcade Fighter Stick dongle GAME STICK 4K LITE SEGAM M8 V7.0

foto

The HUB USB 3.0 + USB-C adapter + dongle do Y3 Joypad with Receiver no-name / Y3 Joypad with GAME STICK 4K LITE SEGAM M8 V4.0 

foto

The HUB USB 3.0 + USB-C adapter + Xbox One X/S Controller

foto

Test 2:

In this test I deleted log 1 and re-did the steps.

[config.ini_TEST_02.txt](https://github.com/user-attachments/files/17371893/config.ini_TEST_02.txt)

[log.log_TEST_02.txt](https://github.com/user-attachments/files/17371896/log.log_TEST_02.txt)


The USB-C adapter + Y3 Joypad (Receiver YC5)

foto

The USB 3.0 HUB + USB-C adapter + Y3 Joypad (Receiver YC5)

foto

**Comment by o0Zz** [comment]
Thank you for the all the information and investigations you provided in this ticket. I believe we can close this issue for now, marking it as unresolved. It appears there's some incompatibility between the Switch and this dongle, but it doesn’t seem to be related to the controller’s profile. I simulated your controller on my side using the Wireshark capture you provided, and everything worked properly. This suggests the controller profile is not at fault, but rather there may be an issue at a lower level between the Switch and the USB dongle. The only potential solutions I see are either updating your USB dongle (if possible) or waiting for Nintendo to modify the Switch's behavior at low level which seems unlikely.

**Comment by MJr0019** [comment]
Ok, thanks for all your help so far.

Thank you very much

It's sad that it won't be compatible.
